console/console-init/ui/src/modules/device/utils/utils.ts

```diff
@@ -0,0 +1,29 @@
+import { IDeviceFilterCriteria } from "../components";
+
+const getFilterCriteriaIndex = (
+  criteriaList: IDeviceFilterCriteria[],
+  criteria: IDeviceFilterCriteria
+) => {
+  for (let i = 0; i < criteriaList.length; i++) {
+    if (compareFilterCriteria(criteriaList[i], criteria)) {
+      return i;
+    }
+  }
+  return -1;
+};
+
+const compareFilterCriteria = (
```

Comment:
we have already function for compare object i.e. compareObject.
we don't need one more function for same purpose.